### ISSUE TITLE
ES6 generators styling

### ISSUE THREAD

**Issue by kaievns** [Prismatik/eslint-config-prismatik#26]
So, I hooked in an updated airbnb config and it seems that my personal preferences mismatch airbnb when it comes to how generator functions should look like. Now, i understand that generators are still new tech for us and you might not have a strong opinion yet. But, never the less, i thought I'd ask about your preferences here.

Generally there are two cases:
## Generator _functions_

options are the following:

``` js
function* () {}  // <- airbnb preferred style
function * () {} // <- nikolay's preferred style
function *() {} //
```
## ES6 class/function _shortcuts_

``` js
class Cat extends Animal {
  * sleep() { } // <- airbnb preferred style
  *eat() { } // <- nikolay's preferred style
}
```

so, basically, you've guessed it, it is Nikolay vs. AirBNB deal. and it is okay if you prefer the AirBNB style, i just want to know which one will bug you less in the future, so i'd make adjustments to the config and redbeard generated code.

**Comment by s-taylor** [comment]
I prefer Airbnb style for functions, less superfluous white space and the \* being attributed (next to) the function itself makes sense in my brain.

Class/function I prefer the Nikolay style.

**Comment by nwinch** [comment]
There's a fourth option for standalone functions:

``` js
function*() {}
```

As we currently use functions with no space, I feel it would make sense to squeeze an asterisk in between.

If I had to choose between the given options, I'd go with airbnb.

For class methods, I prefer:

``` js
*eat() {}
```

**Comment by gilmoreorless** [comment]
I prefer to have a good distinction between generator and non-generator functions. Whitespace is not a dirty word, and it can enhance readability (so I disagree with Simon's "superfluous" assessment).

I find the "no whitespace" options harder to distinguish, because with the asterisk nestled up to the `function` keyword my brain often misses the fact that it's there.
That's just for the function definitions, though, because the asterisk is at the end. Class methods I'm fine with because the asterisk is at the start.

So, in summary, my preference is Nikolay-style for functions, and no real preference for class methods.

**Comment by lyntco** [comment]
I prefer what Nath has suggested - no spaces.

**Comment by charlotteisabella** [comment]
I second what Nathan said

**Comment by kaievns** [comment]
thanks everyone! so i'm closing this with the following:

1) class (method) names are pretty much uniformly preferred Nikolay's style
2) function styles we are at a wide range of preferences, so I'm leaving it in the AirBNB style as a middle ground solution
